enclave-runtime/src/rpc/author/mod.rs

```diff
@@ -180,38 +176,26 @@ where
 		let best_block_hash = Default::default();
 
 		match submission_mode {
-			TopSubmissionMode::Submit => self.execute_on_top_pool(
-				|t: &TopPoolGetter::TrustedOperationPool| -> PoolFuture<TxHash<TopPoolGetter::TrustedOperationPool>, RpcError> {
-					Box::pin(
-						t.submit_one(
-							&generic::BlockId::hash(best_block_hash),
-							TX_SOURCE,
-							stf_operation,
-							shard,
-						)
-						.map_err(map_top_error::<TopPoolGetter::TrustedOperationPool>),
+			TopSubmissionMode::Submit => Box::pin(
+				self.top_pool
+					.submit_one(
+						&generic::BlockId::hash(best_block_hash),
+						TX_SOURCE,
+						stf_operation,
+						shard,
 					)
-				},
-				|e| -> PoolFuture<TxHash<TopPoolGetter::TrustedOperationPool>, RpcError> {
-						Box::pin(ready(Err(e.into())))
-					},
+					.map_err(map_top_error::<TopPool>),
```

Comment:
Was able to revert this part again, not needing the `execute_on_top_pool` method anymore, since we now have plain access to the top pool, not wrapped into a mutex anymore.